flink-connector-mongodb-cdc/src/main/java/com/ververica/cdc/connectors/mongodb/internal/MongoDBConnectorSourceTask.java

```diff
@@ -179,6 +183,7 @@ public List<SourceRecord> poll() throws InterruptedException {
                 }
             }
         }
+        outSourceRecords.add(0, preMetricRecord);
```

Comment:
This will shift all existed records of  outSourceRecords, we can add the preMetricRecord first